internal/processor/kruize_api.go

```diff
@@ -2,169 +2,108 @@ package processor
+	if err != nil {
+		return fmt.Errorf("unable to create payload: %v", err)
+	}
 
-			list_of_experiments = append(list_of_experiments, map[string]string{
-				"experiment_name": experiment_name,
-				"deployment_name": deployment,
-				"namespace":       namespace,
-			})
-			wrapper := []map[string]interface{}{
-				payload_data,
-			}
+	// Update metrics to kruize experiment
+	url := cfg.KruizeUrl + "/updateResults"
 
-			// Update metrics to kruize experiment
-			url := cfg.KruizeUrl + "/updateResults"
-			postBody, err := json.Marshal(wrapper)
-			if err != nil {
-				log.Errorf("unable to marshal payload to json: %v", err)
-				continue
-			}
-			res, err := http.Post(url, "application/json", bytes.NewBuffer(postBody))
-			if err != nil {
-				log.Errorf("An Error Occured while sending metrics: %v", err)
-				continue
-			}
-			if res.StatusCode == 201 {
-				log.Info("Metrics uploaded successfully")
-			} else {
-				defer res.Body.Close()
-				body, _ := io.ReadAll(res.Body)
-				resdata := map[string]interface{}{}
-				if err := json.Unmarshal(body, &resdata); err != nil {
-					log.Errorf("can not unmarshal response data: %v", err)
-					continue
-				}
-				if strings.Contains(resdata["message"].(string), "already contains result for timestamp") {
-					log.Info(resdata["message"])
-				}
-			}
+	res, err := http.Post(url, "application/json", bytes.NewBuffer(payload_data))
+	if err != nil {
+		return fmt.Errorf("an Error Occured while sending metrics: %v", err)
+	}
+	if res.StatusCode == 201 {
+		log.Info("Metrics uploaded successfully")
+	} else {
+		defer res.Body.Close()
+		body, _ := io.ReadAll(res.Body)
+		resdata := map[string]interface{}{}
+		if err := json.Unmarshal(body, &resdata); err != nil {
+			return fmt.Errorf("can not unmarshal response data: %v", err)
+		}
+		if strings.Contains(resdata["message"].(string), "already contains result for timestamp") {
+			log.Info(resdata["message"])
 		}
 	}
-	return list_of_experiments
-}
 
-func list_recommendations(experiment map[string]string) error {
-	error_string := "Failed while listing recommendations from kruize"
-	params := map[string]string{
-		"experiment_name": experiment["experiment_name"],
-		"deployment_name": experiment["deployment_name"],
-		"namespace":       experiment["namespace"],
-	}
+	return nil
+}
 
-	// list recommendation from kruize
+func list_recommendations(experiment string) error {
 	url := cfg.KruizeUrl + "/listRecommendations"
 	client := &http.Client{}
 	req, err := http.NewRequest("GET", url, nil)
 	if err != nil {
-		log.Errorf("An Error Occured %v", err)
-		return errors.New(error_string)
+		return fmt.Errorf("an Error Occured %v", err)
 	}
 	q := req.URL.Query()
-	q.Add("experiment_name", params["experiment_name"])
-	q.Add("deployment_name", params["deployment_name"])
-	q.Add("namespace", params["namespace"])
+	q.Add("experiment_name", experiment)
+	req.URL.RawQuery = q.Encode()
 	res, err := client.Do(req)
```

Comment:
Is the intention here to have recommendations saved to DB post the Recommender system is introduced?

Reply:
Yes, Once recommender is implemented we will save the data coming from listRecommendation API to ROS-OCP DB.